sdcm/cluster.py

```diff
@@ -3768,23 +3768,31 @@ def update_scylla_packages(node, _queue):
             node.remoter.run('tar -xvf /tmp/scylla/*.tar.gz -C /tmp/scylla/', ignore_status=True, verbose=True)
 
             # replace the packages
-            self.log.debug(f'Node distro is {node.distro}, will check the package suffix received')
+            node.log.debug(f'Node distro is {node.distro}, will check the package suffix received')
             if node.distro.is_rhel_like:
                 check_package_suites_distro(node, 'rpm')
-                self.log.info('Installed RPMs before replacing with new RPM files')
+                node.log.info('Installed RPMs before replacing with new RPM files')
                 node.remoter.run('yum list installed | grep scylla', verbose=True)
-                self.log.debug('Will replace the RPM files with the ones provided as a parameter to the test')
+                node.log.debug('Will replace the RPM files with the ones provided as a parameter to the test')
                 node.remoter.sudo('rpm -URvh --replacefiles /tmp/scylla/*.rpm', ignore_status=False, verbose=True)
-                self.log.info('Installed RPMs after replacing with new RPM files')
+                node.log.info('Installed RPMs after replacing with new RPM files')
                 node.remoter.run('yum list installed | grep scylla', verbose=True)
             elif node.distro.is_ubuntu:
                 check_package_suites_distro(node, 'deb')
-                self.log.info('Installed .deb packages before replacing with new .DEB files')
-                node.remoter.run('apt list --installed | grep scylla', verbose=True)
-                node.remoter.sudo('sh -c \'yes Y | dpkg --force-depends -i /tmp/scylla/scylla*\'',
-                                  ignore_status=False, verbose=True)
-                self.log.info('Installed .deb packages after replacing with new .DEB files')
-                node.remoter.run('apt list --installed | grep scylla', verbose=True)
+
+                def list_installed_packages():
+                    node.remoter.sudo('dpkg -l | grep scylla', verbose=True)
```

Comment:
There is `BaseNode.scylla_packages_installed` property. Can you use it here?

And if not, why do you need `sudo` here?

Reply:
fixed